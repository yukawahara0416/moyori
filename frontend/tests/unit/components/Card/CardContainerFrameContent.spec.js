import { shallowMount } from '@vue/test-utils'
import Component from '@/components/Card/CardContainerFrameContent.vue'

let wrapper
let propsData

beforeEach(() => {
  propsData = {
    spot: { marker: { name: 'test' }, data: { id: 1 } },
    id: 1,
    type: 'map'
  }

  wrapper = shallowMount(Component, {
    propsData
  })
})

afterEach(() => {
  wrapper.destroy()
})

describe('proprs', () => {
  it('spot', () => {
    expect(wrapper.props().spot).toStrictEqual(propsData.spot)
    expect(wrapper.props().spot instanceof Object).toBe(true)
  })
  it('type', () => {
    expect(wrapper.props().type).toStrictEqual(propsData.type)
    expect(typeof wrapper.vm.$props.type).toBe('string')
  })
})

describe('template', () => {
  it('snapshot', () => {
    expect(wrapper.vm.$el).toMatchSnapshot()
  })
})
