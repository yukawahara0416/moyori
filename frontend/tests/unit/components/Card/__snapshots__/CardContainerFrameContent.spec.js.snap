// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`template snapshot 1`] = `
<v-row-stub
  class="pa-0"
  tag="div"
>
  <card-container-frame-content-image-stub
    spot="[object Object]"
  />
   
  <v-col-stub
    class="py-0"
    cols="7"
    md="12"
    tag="div"
  >
    <card-container-frame-content-text-stub
      spot="[object Object]"
    />
     
    <card-container-frame-content-action-stub
      id="1"
      spot="[object Object]"
      type="map"
    />
  </v-col-stub>
</v-row-stub>
`;
